### PR TITLE
feat(flamegraph): add view component

### PR DESCRIPTION
**Pull request by JonasBa**
Add some boilerplate and change the flamegraph constructor signature to used the object params so it's easier to read and less err prone

### CODE REVIEW

File: static/app/utils/profiling/flamegraph/useFlamegraphPreferences.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  sorting: 'left heavy' | 'call order';
+  view: 'top down' | 'bottom up';
+}
+
+type FlamegraphPreferencesAction =
+  | {type: 'set color coding'; value: FlamegraphPreferences['colorCoding']}
+  | {type: 'set sorting'; value: FlamegraphPreferences['sorting']}
+  | {type: 'set view'; value: FlamegraphPreferences['view']};
+
+function FlamegraphPreferencesReducer(
```
Comment (Zylphrex) [inline static/app/utils/profiling/flamegraph/useFlamegraphPreferences.ts:18]: nit: 
```suggestion
function flamegraphPreferencesReducer(
```

File: static/app/components/profiling/Flamegraph.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+          <FlamegraphOptionsMenu
+            colorCoding={colorCoding}
+            onColorCodingChange={c => dispatch({type: 'set color coding', value: c})}
+            canvasPoolManager={canvasPoolManager}
+          />
+        </FlamegraphToolbar>
+        <div style={{height: 100, position: 'relative'}}>
+          <FlamegraphZoomViewMinimap
+            flamegraph={flamegraph}
+            highlightRecursion={colorCoding === 'by recursion'}
```
Comment (Zylphrex) [inline static/app/components/profiling/Flamegraph.tsx:109]: Thoughts on removing this `highlightRecursion` prop?
1. it's not being used internally by `FlamegraphZoomViewMinimap` or `FlamegraphZoomView`
2. the `colorCoding` prop already contains the same information (despite not being used at the moment)

File: static/app/components/profiling/Flamegraph.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  useEffect(() => {
+    setFlamegraph(
+      new FlamegraphModel(
+        props.profiles.profiles[0],
+        0,
+        view === 'bottom up',
+        sorting === 'left heavy'
+      )
+    );
+  }, [view === 'bottom up', sorting === 'left heavy']);
```
Comment (Zylphrex) [inline static/app/components/profiling/Flamegraph.tsx:71]:
```suggestion
  }, [props.profiles, view, sorting]);
```

File: static/app/components/profiling/Flamegraph.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      setFlamegraph(
+        new FlamegraphModel(
+          props.profiles.profiles[index],
+          index,
+          view === 'bottom up',
+          sorting === 'left heavy'
+        )
+      );
+    },
+    [view === 'bottom up', sorting === 'left heavy']
```
Comment (Zylphrex) [inline static/app/components/profiling/Flamegraph.tsx:59]:
```suggestion
    [props.profiles, view, sorting]
```

File: static/app/components/profiling/Flamegraph.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      setFlamegraph(
+        new FlamegraphModel(
+          profile.profiles[0],
+          0,
+          view === 'bottom up',
+          sorting === 'left heavy'
+        )
+      );
+    },
+    [view === 'bottom up', sorting === 'left heavy']
```
Comment (Zylphrex) [inline static/app/components/profiling/Flamegraph.tsx:45]:
```suggestion
    [view, sorting]
```

File: static/app/components/profiling/Flamegraph.tsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    <FlamegraphThemeProvider>
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          height: `100vh`,
+          overflow: 'hidden',
+          overscrollBehavior: 'contain',
+        }}
+      >
```
Comment (Zylphrex) [inline static/app/components/profiling/Flamegraph.tsx:83]: Styled components?

File: static/app/components/profiling/Flamegraph.tsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+        <FlamegraphToolbar>
+          <FlamegraphViewSelectMenu
+            view={view === 'bottom up' ? 'bottom up' : 'top down'}
+            sorting={sorting === 'left heavy' ? 'left heavy' : 'call order'}
```
Comment (Zylphrex) [inline static/app/components/profiling/Flamegraph.tsx:87]: Is the ternary here necessary?

**Review by Zylphrex — APPROVED** [review]

File: static/app/components/profiling/Flamegraph.tsx (hunk 2 above)
Comment (JonasBa) [reply static/app/components/profiling/Flamegraph.tsx:109]: yeah, lets remove it 100%

File: static/app/components/profiling/Flamegraph.tsx (hunk 7 above)
Comment (JonasBa) [reply static/app/components/profiling/Flamegraph.tsx:87]: Nope, 🔪

**Pull request merged**